### PR TITLE
Increase spacefinder minAbove for articles without main image, or with a showcase image

### PR DESCRIPTION
**Pull request by Jakeii**
## What does this change?
These 2 types of articles have an issue where inline2 is overlapping most viewed, increasing minAbove for them will prevent it happening.

Also added a line to show minAbove in spacefinder debug.

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## Screenshots

| Before      | After      |
|-------------|------------|
| Screenshot 2022-07-21 at 10 28 21 | Screenshot 2022-07-21 at 10 31 25
 |

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/article-body-adverts.ts [hunk 1]
```diff
@@ -147,11 +152,22 @@ const addDesktopInlineAds = (isInline1: boolean): Promise<boolean> => {
 		filter: filterNearbyCandidates(adSizes.mpu.height),
 	};
 
+	let minAbove = 1000;
```
Comment (domlander) [inline static/src/javascripts/projects/commercial/modules/article-body-adverts.ts:155]: What do you think about using a ternary here?

`let minAbove = isPaidContent ? 1600 : 1000`

**Review by simonbyford — APPROVED** [review]
Nicely done 😀

File: static/src/javascripts/projects/commercial/modules/article-body-adverts.ts (hunk 1 above)
Comment (domlander) [reply static/src/javascripts/projects/commercial/modules/article-body-adverts.ts:155]: If you'd rather not, that's absolutely fine!

Comment (simonbyford) [reply static/src/javascripts/projects/commercial/modules/article-body-adverts.ts:155]: I like the way you've done it! It was ok before when there was only one conditional. Now there's several I find it easier to parse without involving ternaries.

Comment (simonbyford) [reply static/src/javascripts/projects/commercial/modules/article-body-adverts.ts:155]: Small thing but have you tested it on a paid content article?

File: static/src/javascripts/projects/common/modules/spacefinder.ts [hunk 2]
```diff
@@ -87,6 +87,24 @@ const defaultOptions: SpacefinderOptions = {
 	debug: false,
 };
 
+const debugMinAbove = (body: HTMLElement, minAbove: number): void => {
```
Comment (domlander) [inline static/src/javascripts/projects/common/modules/spacefinder.ts:90]: I feel like this should live in the `mark-candidates` file, as it's a part of debugging spacefinder, but then again the minAbove red line isn't a _candidate_ so maybe shouldn't live there.

@simonbyford Would it make sense to rename the `mark-candidates` file to `spacefinder-debug-tool`? Then we could have everything related to debugging spacefinder in the same place.

Comment (Jakeii) [reply static/src/javascripts/projects/common/modules/spacefinder.ts:90]: oh yeah, I like this idea

File: static/src/javascripts/projects/commercial/modules/article-body-adverts.ts (hunk 1 above)
Comment (domlander) [reply static/src/javascripts/projects/commercial/modules/article-body-adverts.ts:155]: Yeah that's a good point. It's probably easier to add more conditions this way too.

Comment (domlander) [reply static/src/javascripts/projects/commercial/modules/article-body-adverts.ts:155]: We have tested on a few. I couldn't find one where the Most Viewed island loads though

Comment (Jakeii) [reply static/src/javascripts/projects/commercial/modules/article-body-adverts.ts:155]: Good point @simonbyford will check that!

**Review by domlander — APPROVED** [review]

File: static/src/javascripts/projects/commercial/modules/article-body-adverts.ts [hunk 3]
```diff
@@ -147,11 +152,22 @@ const addDesktopInlineAds = (isInline1: boolean): Promise<boolean> => {
...
+	if (!hasImages || hasShowcaseMainElement) {
+		minAbove += 100;
+	}
+
 	// For any other inline
 	const relaxedRules: SpacefinderRules = {
 		bodySelector: articleBodySelector,
 		slotSelector: ' > p',
-		minAbove: isPaidContent ? 1600 : 1000,
+		minAbove,
```
Comment (arelra) [inline static/src/javascripts/projects/commercial/modules/article-body-adverts.ts:170]: Is it possible to add to the `selectors` in order to conditionally change `minAbove` depending on the elements in the dom?

Or are they for something else?

Comment (Jakeii) [reply static/src/javascripts/projects/commercial/modules/article-body-adverts.ts:170]: I think they're for setting `min(Above|Below)` from elements in the article body?

**Pull request merged**